src/realm/object-store/sync/sync_session.cpp

```diff
@@ -885,16 +892,30 @@ void SyncSession::create_sync_session()
     }
 
     {
-        std::string sync_route = m_sync_manager->sync_route();
+        // At this point, the sync_route should be valid
+        auto sync_route = m_sync_manager->sync_route();
+        // If sync_route is not valid, it means the location update failed - pass the error up via the registered
+        // error handler
+        if (!sync_route) {
+            Status result{ErrorCodes::LocationUpdateFailed, "Failed to update location prior to sync session start"};
```

Comment:
There isn't visibility into what error occurred at this point... you would have to look at the logs to see what happened. I can make sure an error log message is printed when the location update fails.

Reply:
I don't think the `BadServerUrl` applies, since it's not necessarily due to a malformed URL, but rather the location request failed, which could include a malformed `base_url`, among many other reasons.
I could reuse the `SyncConnectFailed` error code and use the same reason string. The location update failure will be printed as a `logger->error()` when it occurs.